### PR TITLE
run teacher/student in same jit function for performance.

### PR DESCRIPTION
**Pull request by entrpn**
# Description

Runs the teacher/student's forward pass in the same jitted function, removing the d2h and h2d activation transfers and thus improving performance.

# Tests

Updated tests under train_distill_test

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/trainers/post_train/distillation/train_distill.py [hunk 1]
```diff
@@ -173,13 +173,82 @@ def _log_config_details(config: pyconfig.HyperParameters, label: str) -> None:
...
-class MaxTextDistillationTrainer(distillation_trainer.DistillationTrainer):
+class MaxTextDistillationTrainer(peft_trainer.PeftTrainer):
   """Custom Trainer to preserve MaxText fields and log Teacher metrics.
 
   This class overrides `_prepare_inputs` to ensure MaxText-specific fields
   (positions, segment_ids) are passed to the model.
   """
 
+  def __init__(self, student_model, teacher_model, strategy, optimizer, training_config, **kwargs):
+    super().__init__(model=(student_model, teacher_model), optimizer=optimizer, training_config=training_config, **kwargs)
```
Comment (vlad-karp) [inline src/maxtext/trainers/post_train/distillation/train_distill.py]: can a tuple be a cause of an error in the parent PeftTrainer class when the model attribute is handled like a model? Since it is typed like "model: nnx.Module," and we send a tuple here.

**Comment by vlad-karp** [comment]
In general, will this change make it easy to adopt the offline-teacher mode?

File: src/maxtext/trainers/post_train/distillation/train_distill.py (hunk 1 above)
Comment (entrpn) [reply src/maxtext/trainers/post_train/distillation/train_distill.py]: This is a really good point. Jax is treating it like a pytree and that's why its not crashing, but I agree its not a good way to implement this. Let me see if I can update the tunix side to support this better.

**Comment by entrpn** [comment]
> In general, will this change make it easy to adopt the offline-teacher mode?

yes it would. We would just not call the forward fn of the teacher if the teacher's outputs already exist.

File: src/maxtext/trainers/post_train/distillation/train_distill.py (hunk 1 above)
Comment (entrpn) [reply src/maxtext/trainers/post_train/distillation/train_distill.py]: After taking a look at this, there is no current easier way to implement this in tunix. For now, I suggest we keep the implementation as is and revisit in the future. Because we override most methods from PeftTrainer, this solution stays viable.
